### ISSUE TITLE
TypeError: cannot concatenate 'str' and 'NoneType' objects

### ISSUE THREAD

**Issue by sharifyounes** [braintree/braintree_python#109]
When I do `subscription.find(user.subscription_id)`, I get this error if `user.subscription_id==None`. I can of course change my code to:
```
if user.subscription_id:
    subscription.find(user.subscription_id)
```
so no real issue here :)

The "offending" code is here:
https://github.com/braintree/braintree_python/blob/master/braintree/subscription_gateway.py#L38

And there's possibly a similar issue here (though it might be wise not to `repr(nonce)`, I'm not sure):
https://github.com/braintree/braintree_python/blob/master/braintree/credit_card_gateway.py#L58

Suggesting those should be, respectively,

* `raise NotFoundError("subscription with id " + repr(subscription_id) + " not found")`
* `raise NotFoundError("payment method with nonce " + repr(nonce) + " locked, consumed or not found")`

Tests:

For `test_subscription.py`:

```
    @raises(NotFoundError)
    def test_finding_none_raises_not_found_exception(self):
        Subscription.find(None)
```

For `test_credit_card.py`:
```
    @raises(NotFoundError)
    def test_from_nonce_as_none_raises_not_found_exception(self):
        CreditCard.from_nonce(None)
```

**Comment by crookedneighbor** [comment]
I looked through the other server side sdks, and most of them do implement checks around the id existing, so we do that here too. Thanks for the report.

**Comment by crookedneighbor** [comment]
I believe this is fixed now. Closing.

**Comment by crookedneighbor** [comment]
Oh wait, it has not. Re-opening.

**Comment by crookedneighbor** [comment]
Opened a PR internally for this, once it gets merged in and released, I'll close this issue.

**Comment by crookedneighbor** [comment]
This is fixed in 3.55.0
